src/Raven.Server/ServerWide/Commands/Indexes/PutIndexesCommand.cs

```diff
@@ -40,19 +43,48 @@ public PutIndexesCommand(string databaseName, string source, DateTime createdAt,
 
         public override void UpdateDatabaseRecord(DatabaseRecord record, long etag)
         {
-
             if (Static != null)
             {
+                HashSet<string> indexNames = null;
+                HashSet<string> safeFileSystemIndexNames = null;
+
                 foreach (var definition in Static)
+                {
+                    if (record.Indexes.TryGetValue(definition.Name, out _) == false)
+                    {
+                        // this is not an update to an existing index. we'll check for:
+                        // - directory name collisions
+                        // - index name case sensitivity
+
+                        safeFileSystemIndexNames ??= record.Indexes.Select(x => IndexDefinitionBaseServerSide.GetIndexNameSafeForFileSystem(x.Value.Name)).ToHashSet(StringComparer.OrdinalIgnoreCase);
```

Comment:
where this change is coming from? it looks also as a duplicated code, no test as well

Reply:
Actually there were 3 tests that failed.
`CreateFullAndIncrementalBackupWithIndexInTheMiddle`
`CreateIndexWithTheSameLowercasedNameShouldThrow`
`CannotPutIndexWithNameClash`

That is the reason that this change was introduced.

Reply:
ok, but still you need to centrailize this, now you have 2 usages, one in AbstractIndexCreateController and 2nd one here.